modules/features/shared/src/main/java/au/com/shiftyjelly/pocketcasts/shared/NetworkConnectionStatusProvider.kt

```diff
@@ -0,0 +1,15 @@
+package au.com.shiftyjelly.pocketcasts.shared
+
+import android.net.NetworkCapabilities
+import au.com.shiftyjelly.pocketcasts.crashlogging.ConnectionStatusProvider
+import au.com.shiftyjelly.pocketcasts.repositories.playback.NetworkConnectionWatcher
+import javax.inject.Inject
+
+class NetworkConnectionStatusProvider @Inject constructor(
```

Comment:
I didn't add unit tests for this class, as I don't think there's a lot of value - we'd be testing Android framework (if instrumented) or very basic mock (if mocked).